src/scheduler/task.rs

```diff
@@ -559,6 +565,15 @@ impl BlockedTaskQueue {
 				first_task = false;
 				cursor.move_next();
 			}
+
+			// No, then just insert it at the end of the list.
+			self.list.push_back(new_node);
+
+			// If this is the new first task in the list, update the One-Shot Timer
+			// to fire when this task shall be woken up.
```

Comment:
Can you change the comment so it is clear that you are looking at the task with the earliest deadline?

Reply:
yes, see below